arch/arm64/boot/dts/freescale/imx8mp-leica-beluga-0-5.dts

```diff
@@ -0,0 +1,866 @@
+	role-switch-default-mode = "none";
+	snps,dis-u1-entry-quirk;
+	snps,dis-u2-entry-quirk;
+	status = "okay";
+
+	port {
+		usb3_drd_sw: endpoint {
+			remote-endpoint = <&typec_dr_sw>;
+		};
+	};
+};
+
+/* USB 2.0 to hub */
+&usb3_phy1 {
+	status = "okay";
+};
+
+&usb3_1 {
+	status = "okay";
+};
+
+&usb_dwc3_1 {
+	dr_mode = "host";
+	status = "okay";
+};
+
+/* WLAN */
+&usdhc2 {
+	assigned-clocks = <&clk IMX8MP_CLK_USDHC2>;
+	assigned-clock-rates = <400000000>;
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>;
+	pinctrl-1 = <&pinctrl_usdhc2_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc2_200mhz>;
+	no-sd;
+	non-removable;
+	bus-width = <4>;
+	status = "okay";
+};
+
+/* EMMC */
+&usdhc3 {
+	assigned-clocks = <&clk IMX8MP_CLK_USDHC3>;
+	assigned-clock-rates = <400000000>;
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc3>;
+	pinctrl-1 = <&pinctrl_usdhc3_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc3_200mhz>;
+	bus-width = <8>;
+	non-removable;
+	status = "okay";
+};
+
+&wdog1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdog>;
+	fsl,ext-reset-output;
+	status = "okay";
+};
+
+&gpio2 {
+	gpio-line-names =
+		"", "", "PMIC_IRQ_B", "IMX_FPGA_PCIe_xRST", "",
+		"", "", "FPGA_IMX_SYNC", "EFI_CPU_SHDWN", "",
+		"", "IMX_CPU_READY", "", "", "",
+		"", "", "", "", "";
+};
+
+&gpio3 {
+	gpio-line-names =
+		"", "", "", "", "",
+		"", "", "", "", "",
+		"", "", "", "", "",
+		"", "", "", "", "",
+		"", "", "", "IMX_CTS1_xIRQ", "IMX_CTS2_xIRQ",
+		"", "", "EFI_IMX_SPARE", "IMX_KDI2_IRQ", "KDI1_IRQ";
+};
+
+/*
+ * &sai2 {
+ *	#sound-dai-cells = <0>;
+ *	pinctrl-names = "default";
+ *	pinctrl-0 = <&pinctrl_sai2>;
+ *	assigned-clocks = <&clk IMX8MP_CLK_SAI2>;
+ *	assigned-clock-parents = <&clk IMX8MP_AUDIO_PLL1_OUT>;
+ *	assigned-clock-rates = <12288000>;
+ *	status = "okay";
+ * };
+ */
+
+&iomuxc {
+	pinctrl_gpio: gpiogrp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SD1_DATA1__GPIO2_IO03	0x106	/* IMX_FPGA_PCIe_xRST */
+			MX8MP_IOMUXC_SD1_DATA5__GPIO2_IO07	0x106	/* FPGA_IMX_SYNC */
+			MX8MP_IOMUXC_SD1_DATA6__GPIO2_IO08	0x106	/* EFI_CPU_SHDWN */
+			MX8MP_IOMUXC_SD1_STROBE__GPIO2_IO11	0x106	/* IMX_CPU_READY */
+			MX8MP_IOMUXC_SAI5_RXD2__GPIO3_IO23	0x106	/* IMX_CTS1_xIRQ */
+			MX8MP_IOMUXC_SAI5_RXD3__GPIO3_IO24	0x106	/* IMX_CTS2_xIRQ */
+			MX8MP_IOMUXC_HDMI_DDC_SDA__GPIO3_IO27	0x106	/* EFI_IMX_SPARE */
+			MX8MP_IOMUXC_HDMI_CEC__GPIO3_IO28	0x106	/* IMX_KDI2_IRQ */
+			MX8MP_IOMUXC_HDMI_HPD__GPIO3_IO29	0x106	/* IMX_KDI1_IRQ */
+			MX8MP_IOMUXC_SAI1_RXFS__GPIO4_IO00	0x106	/* TD_PPS */
+			MX8MP_IOMUXC_SAI1_RXD7__GPIO4_IO09	0x106	/* IMX_TPM_LP */
+			MX8MP_IOMUXC_SAI2_RXFS__GPIO4_IO21	0x106	/* IMX_SDIO2_INT */
+			MX8MP_IOMUXC_SAI3_RXC__GPIO4_IO29	0x106	/* IMX_FPGA_PROG_EN */
+			MX8MP_IOMUXC_SAI3_TXFS__GPIO4_IO31	0x106	/* FPGA_IMX_IRQ */
+			MX8MP_IOMUXC_SAI1_TXD6__GPIO4_IO18	0x106	/* IMX_FPGA_BT_SYNC */
+		>;
+	};
+
+	pinctrl_eqos: eqosgrp {
+		fsl,pins = <
+			/* IMX_ENET_QOS_MDC */
```

Comment:
for crocodile i kept the line names found in the schematic (?) as a comment behind the hexadecimal pin number

Reply:
@js731ca, Yes, I notesed, I put some comments above the line due to the line length limitation, otherwise can not pass checkpatch.

Reply:
*sigh* you're right...